lib/Doctrine/DBAL/Connection.php

```diff
@@ -1013,6 +1014,32 @@ public function fetchAllKeyValue(string $query, array $params = [], array $types
         return $data;
     }
 
+    /**
+     * Prepares and executes an SQL query and returns the result as an associative array with the keys mapped
+     * to the first column and the values being an associative array representing the rest of the columns
+     * and their values.
+     *
+     * @param string                                           $query  SQL query
+     * @param array<int, mixed>|array<string, mixed>           $params Query parameters
+     * @param array<int, int|string>|array<string, int|string> $types  Parameter types
+     *
+     * @return array<mixed,array<string,mixed>>
+     *
+     * @throws Exception
+     */
+    public function fetchAllAssociativeIndexed(string $query, array $params = [], array $types = []): array
+    {
+        $stmt = $this->executeQuery($query, $params, $types);
+
+        $data = [];
+
+        foreach ($stmt->fetchAll(FetchMode::ASSOCIATIVE) as $row) {
+            $data[array_shift($row)] = $row;
```

Comment:
Is it possible to obtain a float from that call to `array_shift`? If yes, should we assert the result of the call is a valid array key? Bit weird that SA tools do not complain here… Using a float as a key is possible but has surprising results: https://3v4l.org/E1jGM

Reply:
When floats are used as array keys, they are cast to an integer. I don't think DBAL should do much about that. It can be not only float but NULL. It's up to the developer who uses this API to decide what they are using as a key.